File: answers/other/answers/leetcode/Delete Node in a Linked List.cpp
//@type Linked List
//@result 83 / 83 test cases passed. Status: Accepted Runtime: 16 ms Submitted: 1 minute ago You are here! Your runtime beats 9.05% of cpp submissions.

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    void deleteNode(ListNode* node) {
        if (NULL == node) {
		}
        if (NULL == node->next) {
			node = NULL;
		}
		else {
			node->val = node->next->val;
			node->next = node->next->next;
		}
    }
};

File: answers/leetcode/Delete Node in a Linked List/Delete Node in a Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        if (null == node) {
            return;
        } else if (null == node.next) {
            node = null;
        } else {
            node.val = node.next.val;
            node.next = node.next.next;
        }
    }
}